app/src/main/java/org/mozilla/focus/widget/EqualDistributeGrid.java

```diff
@@ -0,0 +1,168 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.widget;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.mozilla.focus.R;
+
+/**
+ * A just-work custom ViewGroup for Bottom Sheet. It *ASSUME* each children has the same width and
+ * height, then to layout children in horizontal orientation by inserting equal spacing between each
+ * children. If children number is more than row-capacity(specified in XML), rest children will move
+ * to next row, and align previous row.
+ * <p>
+ * ie: put 12 cells in this ViewGroup with row-capacity = 5, it will look like this
+ * [C C C C C]
+ * [C C C C C]
+ * [C C      ]
+ */
+public class EqualDistributeGrid extends ViewGroup {
+
+    private int rowCapacity = Integer.MAX_VALUE;
+    private int gutter = 0;
+
+    public EqualDistributeGrid(Context context) {
+        super(context);
+    }
+
+    public EqualDistributeGrid(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.EqualDistributeGrid);
+        rowCapacity = a.getInt(R.styleable.EqualDistributeGrid_horizontalCapacity, Integer.MAX_VALUE);
+        gutter = a.getDimensionPixelSize(R.styleable.EqualDistributeGrid_verticalGutter, 0);
+        a.recycle();
+    }
+
+    @Override
+    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
+        measureChildren(widthMeasureSpec, heightMeasureSpec);
+
+        final int visibleCnt = getVisibleCount();
+
+        // how many cells in first row
+        final int rowLength = Math.min(rowCapacity, visibleCnt);
+
+        // to know width and height for each cell
+        final View reference = firstVisibleChild();
+
+        int width = calculateWidth(widthMeasureSpec, heightMeasureSpec, visibleCnt, rowLength, reference);
+        int height = calculateHeight(widthMeasureSpec, heightMeasureSpec, visibleCnt, rowLength, reference);
+        setMeasuredDimension(width, height);
+    }
+
+    private int calculateWidth(int widthMeasureSpec,
```

Comment:
If we dive into the implementation, its logic is not so similar. Moreover, it might become more complex if we want to make it more general someday. For that case, keep two methods would be easier to rewrite, my opnion.